pgmpy/factors/continuous/JointGaussianDistribution.py

```diff
@@ -0,0 +1,335 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import division
+
+import six
+import numpy as np
+from scipy.stats import multivariate_normal
+
+from pgmpy.factors import ContinuousFactor
+
+
+class JointGaussianDistribution(ContinuousFactor):
+    u"""
+    In its most common representation, a multivariate Gaussian distribution
+    over X1...........Xn is characterized by an n-dimensional mean vector μ,
+    and a symmetric n x n covariance matrix Σ.
+    This is the base class for its representation.
+    """
+    def __init__(self, variables, mean, covariance):
+        """
+        Parameters
+        ----------
+        variables: iterable of any hashable python object
+            The variables for which the distribution is defined.
+        mean: n x 1, array like
+            n-dimensional vector where n is the number of variables.
+        covariance: n x n, 2-d array like
+            n x n dimensional matrix where n is the number of variables.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from pgmpy.factors import JointGaussianDistribution as JGD
+        >>> dis = JGD(['x1', 'x2', 'x3'], np.array([[1], [-3], [4]]),
+        ...             np.array([[4, 2, -2], [2, 5, -5], [-2, -5, 8]]))
+        >>> dis.variables
+        ['x1', 'x2', 'x3']
+        >>> dis.mean
+        array([[ 1],
+               [-3],
+               [4]]))
+        >>> dis.covariance
+        array([[4, 2, -2],
+               [2, 5, -5],
+               [-2, -5, 8]])
+        >>> dis.pdf([0,0,0])
+        0.0014805631279234139
+        """
+        no_of_var = len(variables)
+
+        if len(mean) != no_of_var:
+            raise ValueError("Length of mean_vector must be equal to the\
+                                 number of variables.")
+
+        self.mean = np.asarray(np.reshape(mean, (no_of_var, 1)), dtype=float)
+        normal_pdf = lambda *args: multivariate_normal.pdf(args, self.mean.reshape(1, no_of_var)[0],
+                                                           covariance)
+        self.covariance = np.asarray(covariance, dtype=float)
+        self._precision_matrix = None
+
+        if self.covariance.shape != (no_of_var, no_of_var):
```

Comment:
@khalibartan @ankurankan I will look into it. I also forgot to add type
checks for array elements. Will do that too.
On 21 Jun 2016 11:41, "Utkarsh" <notifications@github.com> wrote:

> In pgmpy/factors/continuous/JointGaussianDistribution.py
> <https://github.com/pgmpy/pgmpy/pull/693#discussion_r67812904>:
>
> > +        >>> dis.pdf([0,0,0])
> > +        0.0014805631279234139
> > +        """
> > +        no_of_var = len(variables)
> > +
> > +        if len(mean) != no_of_var:
> > +            raise ValueError("Length of mean_vector must be equal to the\
> > +                                 number of variables.")
> > +
> > +        self.mean = np.asarray(np.reshape(mean, (no_of_var, 1)), dtype=float)
> > +        normal_pdf = lambda *args: multivariate_normal.pdf(args, self.mean.reshape(1, no_of_var)[0],
> > +                                                           covariance)
> > +        self.covariance = np.asarray(covariance, dtype=float)
> > +        self._precision_matrix = None
> > +
> > +        if self.covariance.shape != (no_of_var, no_of_var):
>
> @yashu-seth <https://github.com/yashu-seth> @ankurankan
> <https://github.com/ankurankan> I think we should also check whether
> covariance is a symmetric matrix or not. But I don't know any efficient way
> of doing that.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pgmpy/pgmpy/pull/693/files/3fe19353ee1b25a9414338631cb2070659b5dff8#r67812904>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ANikob9Qf6DZqhali5v6iSKDPZr667xYks5qN4CNgaJpZM4Ip9zh>
> .
>